### PR TITLE
chore: add execution_id to any available report logs

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
To improve logging, we're looking to be able to search for an execution ID to tie together logs from the entire alert/report Celery job. To do this, I have added an execution_id to the Screenshot classes (dashboard and chart), the slack and email send methods and the webdriver screenshot method. The report task is already passing the execution id through to the commands, etc, so I also passed the celery execution id through the thumbnail generation to the screenshot logs as well.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/screenshots.py [hunk 1]
```diff
@@ -204,41 +219,30 @@ class ChartScreenshot(BaseScreenshot):
...
 
-    def __init__(
-        self,
-        url: str,
-        digest: str,
-        window_size: Optional[WindowSize] = None,
-        thumb_size: Optional[WindowSize] = None,
-    ):
-        # Chart reports are in standalone="true" mode
+    def __init__(self, url: str, digest: str, *args: ScreenshotDetails):
```
Comment (eschutho) [inline superset/utils/screenshots.py:222]: @betodealmeida what do you think of this as a way to resolve the too-many-arguments linting error? I see that we're ignoring it in a lot of places, but I'd like to find a solution that we can use as a pattern for bundling up extra args/kwargs, and wanted to get your thoughts on this approach. (also see line 228 below). Thanks!

File: superset/utils/webdriver.py [hunk 2]
```diff
@@ -198,10 +202,11 @@ def get_screenshot(
             ]
             logger.debug("Wait %i seconds for chart animation", selenium_animation_wait)
             sleep(selenium_animation_wait)
-            logger.debug(
-                "Taking a PNG screenshot of url %s as user %s",
+            logger.info(
+                "Taking a PNG screenshot of url %s as user %s with execution_id: %s",
```
Comment (eschutho) [inline superset/utils/webdriver.py:206]: @craig-rueda I resolved a conflict here from a recent change of yours. I was hoping to get this execution id into our production logs. Do you think it's something that we should only log locally?

Comment (craig-rueda) [reply superset/utils/webdriver.py:206]: I think this should be `debug`, as it's cluttering things unnecessarily. We can leverage the logging context if you wanted to add `execution_id` to the rendered log's JSON, which will land in the logging aggregator as a field

Comment (eschutho) [reply superset/utils/webdriver.py:206]: Oh, and that's something that you can filter on? That sounds like a better implementation overall than adding it to the message field.

**Pull request closed without merging**